core/src/common.cpp

```diff
@@ -339,33 +339,15 @@ namespace irods::http
 
 	auto map_json_to_user(const nlohmann::json& _json) -> std::optional<std::string>
 	{
-		const auto& oidc_config{irods::http::globals::oidc_configuration()};
-		const static auto user_claim{oidc_config.find("irods_user_claim")};
-		const static auto attribute_mapping{oidc_config.find("user_attribute_mapping")};
+		auto json_res_string{to_string(_json)};
+		auto match_func{irods::http::globals::user_map_lib().get<char* (const char*)>("user_mapper_match")};
+		auto* res{match_func(json_res_string.c_str())};
```

Comment:
Should any of these be `const`, `static`, or both?

Reply:
Given the user mapping plugin is only loaded at program start, `match_func` could be static.

Is the call to `.get` doing a lookup every time?

Reply:
The `.get()` call does a `.has()` call every time, I believe, so it would do a lookup every time.

Reply:
Done.